smbus2/smbus2.py

```diff
@@ -49,6 +50,9 @@
 I2C_SMBUS_I2C_BLOCK_DATA = 8
 I2C_SMBUS_BLOCK_MAX = 32
 
+#FreeBSD RDWR syscall
+I2CRDWR = 0x80106906
```

Comment:
Thank you for your effort.
I cannot use this change on Raspberry PI 3(armv7).
I got the below message[NG].
  File "/tmp/smbus2.py", line 731, in i2c_rdwr
    ioctl(self.fd, I2CRDWR, ioctl_data)
OSError: [Errno 25] Inappropriate ioctl for device

On armv7(Raspberry PI-PI3), it should be below.
I2CRDWR = 0x80086906

We should change this value on the CPU arch/hardware platform.

Reply:
Well, that's odd. I have a couple of systems here, and running the following code:

``` C
#include <stdio.h>
#include <stdint.h>
#include <sys/types.h>
#include <dev/iicbus/iic.h>

int main() {
        printf("%lx\n", I2CRDWR);
        return 0;
}
```

gives me the following results:

``` bash
# uname -a
FreeBSD amd64-system 13.2-RELEASE-p2 FreeBSD 13.2-RELEASE-p2 GENERIC amd64
# ./test
80106906
```

``` bash
# uname -a
FreeBSD arm64-system 13.2-RELEASE-p4 FreeBSD 13.2-RELEASE-p4 GENERIC arm64
# ./test 
80106906
```

unfortunately i don't have an armv7 system here to test on, but your raspi3 will run aarch64 (;
Are you sure you are using the correct device?

best

Reply:
I have three types of Raspberry PI(RPI-zero, RPI3, RPI4).

I tested some devices, including amd64 and cloud.

64bit CPU's I2CRDWR is 80106906
* 13.2-RELEASE-p4 GENERIC amd64(Intel CPU)
* 12.4-RELEASE-p6 amd64(Goolgle Cloud Platform)
* 14.0-CURRENT arm64(Pine64)
* 13.2-RELEASE-p4 GENERIC arm64 (Oracle Cloud)
   (This time, I did not test with RPI4)

32bit CPU's I2CRDWR is 80086906
* 14.0-CURRENT arm (PRI-3)
* 13.2-RELEASE arm  (PRI ZERO)

Reply:
So smth. like this could work:

``` python
import platform
(bits, _) = platform.architecture()
>>> I2CRDWR = { '64bit': 0x80106906, '32bit': 0x80086906 }[bits]
```

Reply:
now i have one question left: whats about i386 systems? they also use `0x80086906` ? I don't have any 32bit systems run anymore. Can anyone here test this?

Reply:
I checked I2CRDWR on i386 systems.
All system's I2CRDWR  is 0x80086906[OK]

0x80086906      
FreeBSD freebsd-i386 13.2-RELEASE FreeBSD 13.2-RELEASE releng/13.2-n254617-525ecfdad597 GENERIC i386
0x80086906
FreeBSD FreeBSD-15-i386 15.0-CURRENT FreeBSD 15.0-CURRENT #0 main-n265904-bb679b0c4909: Thu Oct 12 02:53:29 UTC 2023     root@releng3.nyi.freebsd.org:/usr/obj/usr/src/i386.i386/sys/GENERIC i386